ksqldb-rest-app/src/main/java/io/confluent/ksql/api/auth/AuthenticationPlugin.java

```diff
@@ -46,4 +46,9 @@ public interface AuthenticationPlugin {
   CompletableFuture<Principal> handleAuth(RoutingContext routingContext,
       WorkerExecutor workerExecutor);
 
+
+  default String getAuthToken(final RoutingContext routingContext) {
```

Comment:
Maybe add an additional comment here about how this is different from handleAuth.  This is a little funny because we expose the auth token (as opposed to just treating it as entirely internal to the plugin, as handleAuth does) in addition to exposing `handleAuth`, since we need it for the purpose of forwarding on requests inter-node as using it as a backup header for connect.  Since we need it for those two purposes, I don't think we can easily hide it in a plugin method like `prepareRequest()`.

Also, I assume that since this is default, it shouldn't break any existing plugins.